core/src/main/java/io/aiven/klaw/service/EnvsClustersTenantsControllerService.java

```diff
@@ -704,10 +707,17 @@ public ApiResponse addNewEnv(EnvModel newEnv) throws KlawException {
     env.setEnvExists("true");
 
     try {
+      EnvTag tag =
+          addEnvironmentMapping(
```

Comment:
if it is applicable only for schema envs, we can add that condition here ?

Reply:
The Kafka Env also get checked to make sure if they have an existing associated environment that it doesn't get overridden during an update of the environment details so it does require it to dip in there as well.